File: tp/61818 - López Tisera, Gastón/tp6/frontend/types/purchase.ts
export interface PurchaseItem {
  id: number;
  producto_id: number;
  nombre: string;
  cantidad: number;
  precio_unitario: number;
}

export interface Purchase {
  id: number;
  fecha: string;
  direccion: string;
  tarjeta: string;
  envio: number;
  total: number;
  items: PurchaseItem[];
}

File: tp/61818 - López Tisera, Gastón/tp6/backend/app/schemas/purchase.py
from datetime import datetime

from pydantic import BaseModel


class PurchaseItemRead(BaseModel):
    id: int
    producto_id: int
    nombre: str
    cantidad: int
    precio_unitario: float

    class Config:
        from_attributes = True


class PurchaseRead(BaseModel):
    id: int
    fecha: datetime
    direccion: str
    tarjeta: str
    envio: float
    total: float
    items: list[PurchaseItemRead]

    class Config:
        from_attributes = True

